ocp_resources/datavolume.py

```diff
@@ -269,29 +274,73 @@ def _check_none_pending_status(self, failure_timeout=120):
                     continue
                 break
         except TimeoutExpiredError:
-            self.logger.error(f"{self.name} status is {sample}")
+            self.logger.error(f"{self.name} status is {sample.status}")
             raise
 
-    def wait_for_dv_success(self, timeout=600, failure_timeout=120):
+    def _check_dv_progress(self):
+        dv_status = self.instance.status
+        dv_status_conditions = dv_status.get("conditions")
+        if (
+            dv_status.restartCount
+            and dv_status.restartCount > 3
+            and dv_status_conditions
+            and dv_status_conditions[-1].type == "Running"
```

Comment:
Why do you rely on the condition order `[-1]` ?

Reply:
we don't needto rely on order, please change it

Reply:
changing it, and moving this function out of here